src/main/java/wallet/command/Command.java

```diff
@@ -0,0 +1,156 @@
+     *
+     * @param fullCommand The full command input by the user
+     * @param taskList    The current list of task
+     * @param fileIO      The class object that handles file IO
+     * @return true if the command given is bye
+     */
+    public static boolean parse(String fullCommand, TaskList taskList, Storage fileIO,
+                                ContactList contactList, RecordList recordList, ExpenseList expenseList) {
+        boolean isExit = false;
+
+        String[] command = fullCommand.split(" ", 2);
+        if (command[0].equals("list")) {
+            if (command[1].equals("task")) {
+                int count = 1;
+                System.out.println("Here are the tasks in your list:");
+                for (Task t : taskList.getTaskList()) {
+                    System.out.println(count + "." + t.toString());
+                    count++;
+                }
+            } else if (command[1].equals("contact")) {
+                int count = 1;
+                System.out.println("Here are the contacts in your list:");
+                for (Contact c : contactList.getContactList()) {
+                    System.out.println(count + "." + c.toString());
+                    count++;
+                }
+            } else if (command[1].equals("expense")) {
+                int count = 1;
+                System.out.println("Here are the expenses in your list:");
+                for (Expense e : expenseList.getExpenseList()) {
+                    System.out.println(count + ". " + e.toString());
+                    count++;
+                }
+            }
+        } else if (command[0].equals("find")) {
+            int count = 1;
+            try {
+                String keyword = command[1];
+                System.out.println("Here are the matching tasks in your list:");
+                for (Task t : taskList.getTaskList()) {
+                    if (t.getDescription().contains(keyword)) {
+                        System.out.println(count + "." + t.toString());
+                        count++;
+                    }
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("☹ OOPS!!! Please specify a keyword to search for.");
+            }
+        } else if (command[0].equals("done")) {
+            try {
+                int num = Integer.parseInt(command[1]) - 1;
+                Task task = taskList.getTask(num);
+                if (task.getStatus()) {
+                    System.out.println("This task is already done.");
+                    System.out.println(task.toString());
+                } else {
+                    task.markAsDone();
+                    taskList.modifyTask(num, task);
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(task.toString());
+                    //fileIO.updateToFile(task, num);
```

Comment:
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.